search_gov_crawler/search_gov_spiders/spiders/domain_spider.py

```diff
@@ -1,9 +1,43 @@
 from scrapy.linkextractors import LinkExtractor
 from scrapy.spiders import CrawlSpider, Rule
+from scrapy.http import Response
 import os
 import sys
+import re
 from ..items import SearchGovSpidersItem
 
+IGNORED_EXTENSIONS = [
+    # archives
+    "7z", "7zip", "bz2", "rar", "tar", "tar.gz", "xz", "zip", "gz"
+    # images
+    "mng", "pct", "bmp", "gif", "jpg", "jpeg", "png", "pst", "psp", "tif", "tiff", "ai", "drw", "dxf", "eps", "ps", "svg", "cdr", "ico",
+    # audio
+    "mp3", "wma", "ogg", "wav", "ra", "aac", "mid", "au", "aiff",
+    # video
+    "3gp", "asf", "asx", "avi", "mov", "mp4", "mpg", "qt", "rm", "swf", "wmv", "m4a", "m4v", "flv", "webm",
+    # office suites
+    "ppt", "pptx", "pps", "odt", "ods", "odg", "odp",
+    # other
+    "css", "exe", "bin", "rss", "dmg", "iso", "apk", "js", "xml", "ibooks", "cfm", "ics", "nc", "prj", "sfx",
+]
+
+ALLOWED_CONTENT_TYPE = [
+    "text/html",
+    "text/plain",
+    "application/msword",
+    "application/pdf",
+    "application/vnd.ms-excel",
+    "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
+    "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+]
+
+def valid_content_type(type_whitelist, content_type_header):
+        content_type_header = str(content_type_header)
+        for type_regex in type_whitelist:
+            if re.search(type_regex, content_type_header):
+                return True
+        return False
+
 starting_urls = os.path.join(
     os.path.dirname(sys.modules[__name__].__file__), "../utility_files/startingUrls.txt"
 )
```

Comment:
i know this piece didnt change but have you considered moving this and the reading of the file into the init method (or into a function that is called during init)?  This functionality seems like its only used if no domain (or url) is passed so there could be times when you are doing this work (and potentially causing an error) when you don't have to.

Also, do you ever use pathlib? for this stuff? I like the API a bit better than os, for instance you could replace reading and parsing the files that you have here with:
```python
starting_urls_file = Path(__file__).parent.parent / "utility_files" / "startingUrls.txt"
start_urls_list = starting_urls_file.resolve().read_text().splitlines()
```

Reply:
Yeah, this looks a lot cleaner, but I didn't want to change too much outside the scope of the ticket. We can create a cleanup/maintenance task and implement it everywhere.